tests/unit/drivers/test_convert_driver.py

```diff
@@ -0,0 +1,28 @@
+import numpy as np
+from PIL import Image
+import pytest
+
+from jina.drivers.convert import Blob2PngURI
+from jina.drivers.helper import array2pb
+from jina.proto import jina_pb2
+
+
+def create_document(arr_size):
+    doc = jina_pb2.Document()
+    doc.blob.CopyFrom(array2pb(np.random.randint(0, 255, arr_size)))
+    return doc
+
+
+docs = [create_document([32*28]), create_document([32, 28]), create_document([32, 28, 3])]
+modes = ['L', 'L', 'RGB']
+test_data = zip(docs, modes)
+
+
+@pytest.mark.parametrize('data', test_data)
+def test_blob2pnguri_driver(data):
+    doc, mode = data
+    width, height = 28, 32
+
+    driver = Blob2PngURI(target='uri', width=width, height=height)
+    driver._apply(doc)
+    Image.frombytes(mode, (width, height), doc.uri.encode()) # just to check if the data is enough for the image recreation
```

Comment:
some assertion at least about the resulting size?

Reply:
I had actually kept the assertion but felt it useless since the shape is given for recreating the image. If it is not able to create it from the byte data, it throws the error. Also, I observed that this is not very reliable for exact checking. I can have input image of some shape and then have this recreated of some other shape and it still works. So this recreation is just to check if the uri data exists and is enough.

Reply:
and testing that URI2Buffer returns the same value as in Blob would make sense?

Reply:
The blob is converted to numpy arr which is then converted to type uint8 in png_converter. I think due to type conversions, we might not be able to assert blobs.
arr = pb2array(d.blob)

The original blob is made from numpy array and the final blob will be made from png after doing URI2Buffer.

Reply:
Ok have u tested in ur example then that it works at least?

Reply:
maybe we can have an integration test for this one

Reply:
Yes tested it for certain values. The test throws error if the uri is empty or there is a huge mismatch of original image shape and final image shape. If original is 32,32 it works for like 32, 64 final but fails for 32, 128. Not able to understand this phenomenon. Throws ValueError: not enough image data.

Let me check integration test. Pushing code for now for testing current.